src/Service.php

```diff
@@ -7,179 +8,315 @@
-                    'nickname'         => $method . ' for ' . $service->getName(),
-                    'type'             => $service->getName(),
-                    'parameters'       => $parameters,
-                    'responseMessages' => $operation->getResponseStatusCodes(),
-                ];
+    protected function getFirstPostRequestDescription()
+    {
+        foreach ($this->getAllOperations() as $operation) {
+            $method = $this->getMethodFromOperation($operation);
+            if ($method === 'post') {
+                return $operation->getRequestDescription();
             }
-            $operationGroups[] = [
-                'operations' => $operations,
-                'path'       => $routeWithReplacements
-            ];
         }
+    }
 
+    protected function getFieldsForDefinitions()
+    {
         // Fields are part of the default input filter when present.
-        $fields = $service->fields;
+        $fields = $this->service->fields;
         if (isset($fields['input_filter'])) {
             $fields = $fields['input_filter'];
         }
+        return $fields;
+    }
 
-        $requiredProperties = $properties = [];
-        foreach ($fields as $field) {
-            $properties[$field->getName()] = [
-                'type' => method_exists($field, 'getFieldType') ? $field->getFieldType() : 'string',
-                'dataType' => method_exists($field, 'getFieldType') ? $field->getFieldType() : 'string',
-                'description' => $field->getDescription()
-            ];
-            if ($field->isRequired()) {
-                $requiredProperties[] = $field->getName();
-            }
+    protected function getFieldProperties(Field $field)
+    {
+        $type = $this->getFieldType($field);
+        $properties =[];
+        $properties['type']=$type;
+        if ($type === self::ARRAY_TYPE) {
+            $properties['items'] = ['type' => self::DEFAULT_TYPE];
         }
+        $properties['description'] =$field->getDescription();
+        return $this->cleanEmptyValues($properties);
+    }
 
-        return [
-            'apiVersion'     => $service->api->getVersion(),
-            'swaggerVersion' => '1.2',
-            'basePath'       => $this->baseUrl,
-            'resourcePath'   => $routeBasePath,
-            'apis'           => $operationGroups,
-            'produces'       => $service->requestAcceptTypes,
-            'models'         => [
-                $service->getName() => [
-                    'id'         => $service->getName(),
-                    'required'   => $requiredProperties,
-                    'properties' => $properties,
-                ],
-            ],
-        ];
+    protected function getFieldType(Field $field)
+    {
+        return (method_exists($field, 'getFieldType') &&
+            !empty($field->getFieldType())) ?
+            $field->getFieldType() : self::DEFAULT_TYPE;
```

Comment:
Re-flow this to have each line have a single semantic:

```php
return method_exists($field, 'getFieldType') && ! empty($field->getFieldType())
    ? $field->getFieldType()
    : self::DEFAULT_TYPE;
```

This makes it easier to understand that the line is a ternary condition, and breaks it into the condition, the condition-matched return, and the unmatched return.